crates/client/src/commander.rs

```diff
@@ -466,16 +473,21 @@ impl Commander {
 
         let cargo_thread = std::thread::spawn(move || -> Result<(), Error> {
             let output = Self::expand_package("program_client");
+            // release progress bar loop here as the expansion is the longest part
+            // further we cannot release after parsing as parsing can panic which will not
+            // release the lock
+            c_mutex.store(false, std::sync::atomic::Ordering::SeqCst);
+            let output = output.expect("Program Client expansion failed");
 
             if output.status.success() {
-                let mut code = c_shared_mutex.lock().unwrap();
+                let mut code = c_shared_mutex
+                    .lock()
+                    .expect("Acquire Program Client lock failed");
                 code.push_str(&String::from_utf8(output.stdout)?);
 
-                c_mutex.store(false, std::sync::atomic::Ordering::SeqCst);
                 Ok(())
             } else {
                 // command failed leave unmodified
```

Comment:
Are you sure, that this is OK? If there was a problem during the expansion, then the String inside the shared_mutex will be empty and this will be then further passed to the get_use_statemets function that will do nothing. So if there is a problem during the expansion, it will fail silently without notifying the user, that in case he had some custom imports, these were not detected and were deleted. 

I know, that we want to get rid of the expansion and also that the user will notice sooner or later due to compilation errors, but anyway... If I am correct, it would be better to fix it.

Reply:
True, check #134 please.